Tests/GRPCHTTP2CoreTests/GRPCStreamStateMachineTests.swift

```diff
@@ -1921,39 +1921,26 @@ final class GRPCStreamServerStateMachineTests: XCTestCase {
 
   func testReceiveMetadataWhenClientOpenAndServerIdle() throws {
     var stateMachine = self.makeServerStateMachine(targetState: .clientOpenServerIdle)
-
-    // Try receiving initial metadata again - should fail
-    XCTAssertThrowsError(
-      ofType: RPCError.self,
-      try stateMachine.receive(headers: .clientInitialMetadata, endStream: false)
-    ) { error in
-      XCTAssertEqual(error.code, .internalError)
-      XCTAssertEqual(error.message, "Client shouldn't have sent metadata twice.")
-    }
+    XCTAssertEqual(
+      try stateMachine.receive(headers: .clientInitialMetadata, endStream: false),
+      .doNothing
+    )
```

Comment:
I think we shouldn't just ignore these: if the client sends metadata more than once we should close the stream instead. My main concern is that otherwise we are vulnerable to bad actors / buggy clients spamming the server indefinitely with metadata packets.